src/pages/index.js

```diff
@@ -60,39 +90,119 @@ editFormValidator.enableValidation();
 
 const addFormValidator = new FormValidator(validationSettings, addCardForm);
 addFormValidator.enableValidation();
-/*
- * =================================================
- *               FUNCTION HANDLERS
- * =================================================
- */
+
+const avatarValidator = new FormValidator(validationSettings, avatarForm);
+avatarValidator.enableValidation();
+
+// FUNCTION HANDLERS
+function handleUpdatingAvatar(inputValues) {
+  avatarPopupForm.setButtonLoading();
+  api
+    .updateAvatar(inputValues)
+    .then((data) => {
+      userInfo.setAvatarImage({
+        avatar: data.avatar,
+      });
+      avatarValidator.resetValidation();
+      avatarPopupForm.close();
+    })
+    .catch((err) => {
+      console.error("error updating profile image", err);
+    })
+    .finally(() => {
+      avatarPopupForm.resetButtonLoading();
+    });
+}
+
 function handleProfileEditSubmit(inputValues) {
-  //get (inputValues) from popupWithForm
-  userInfo.setUserInfo(inputValues);
-  editFormValidator.resetValidation();
-  editPopupForm.close();
+  editPopupForm.setButtonLoading();
+  api
+    .updateProfileInfo(inputValues)
+    .then((userData) => {
+      userInfo.setUserInfo({
+        title: userData.name,
+        description: userData.about,
+      });
+      editFormValidator.resetValidation();
```

Comment:
` editFormValidator.resetValidation();` is not needed here because you can't have possible errors after successful submitting

Please, move it into the open handler